tmt/steps/report/junit.py

```diff
@@ -111,36 +68,112 @@ class ReportJUnitData(tmt.steps.report.ReportStepData):
         help='Include full standard output in resulting xml file.')
 
 
+class ResultSummaryContext:
+    """
+    A class which keeps the results context (especially the result summary) for
+    JUnit template.
+    """
+
+    def __init__(self, results):
+        self._results = results
+
+    def __iter__(self):
+        return iter(self._results)
+
+    def __len__(self):
+        return len(self._results)
+
+    # TODO: Some better filtering function?
+    @functools.cached_property
+    def executed(self):
+        return [r for r in self._results if r.result != ResultOutcome.INFO]
+
+    @functools.cached_property
+    def skipped(self):
+        return [r for r in self._results if r.result == ResultOutcome.INFO]
+
+    @functools.cached_property
+    def failed(self):
+        return [r for r in self._results if r.result == ResultOutcome.FAIL]
+
+    @functools.cached_property
+    def errored(self):
+        return [r for r in self._results if r.result in [
+                ResultOutcome.ERROR,
+                ResultOutcome.WARN]]
+
+    @functools.cached_property
+    def duration(self):
+        return sum([duration_to_seconds(r.duration) for r in self._results])
+
+
 @tmt.steps.provides_method('junit')
 class ReportJUnit(tmt.steps.report.ReportPlugin[ReportJUnitData]):
     """
-    Save test results in JUnit format.
+    Save test results in chosen JUnit flavor format. When flavor is set to
+    custom, the ``template_path`` with a path to a custom template must be
+    provided.
 
     When ``file`` is not specified, output is written into a file
     named ``junit.xml`` located in the current workdir.
     """
 
     _data_class = ReportJUnitData
 
+    def check_options(self) -> None:
+        """ Check the module options """
+
+        if self.data.flavor and self.data.template_path:
+            raise tmt.utils.ReportError("The options '--template-path' and "
+                                        "'--flavor' are mutually exclusive.")
+
+        self.data.flavor = self.data.flavor or DEFAULT_FLAVOR_NAME
+
     def prune(self, logger: tmt.log.Logger) -> None:
         """ Do not prune generated junit report """
 
     def go(self, *, logger: Optional[tmt.log.Logger] = None) -> None:
         """ Read executed tests and write junit """
         super().go(logger=logger)
 
-        junit_xml = import_junit_xml()
-        suite = make_junit_xml(self)
+        self.check_options()
 
         assert self.workdir is not None
         f_path = self.data.file or self.workdir / DEFAULT_NAME
+
+        # Get the template according to chosen flavor
+        if self.data.flavor == 'custom':
+            template_path = Path(self.data.template_path)
```

Comment:
Check `tmt.utils.normalize_path`, it should be used in the field definition, and the type should be `Optional[Path]`. The path-like input should be converted into a `Path` instance on the boundary, and that's the step data class.

Reply:
Is it correct now? Can I normalize the path the same way (with `tmt.utils.normalize_path`) also for the `--file` option?

Reply:
I believe so, `--file` looks like a relict of times when `normalize_path` did not exist yet.